### PR TITLE
Make SimCalorimetry/EcalSimAlgos compile with C++20

### PR DESCRIPTION
**Pull request by iarspider**
#### PR description:

Title says it all

#### PR validation:

Code compiles

### CODE REVIEW

**Comment by iarspider** [comment]
type bugfix

**Comment by iarspider** [comment]
please test

File: SimCalorimetry/EcalSimAlgos/interface/EcalTDigitizer.h [hunk 1]
```diff
@@ -27,9 +27,9 @@ class EcalTDigitizer {
   typedef typename Traits::DigiCollection DigiCollection;
   typedef typename Traits::EcalSamples EcalSamples;
 
-  EcalTDigitizer<Traits>(EcalHitResponse* hitResponse, ElectronicsSim* electronicsSim, bool addNoise);
+  EcalTDigitizer(EcalHitResponse* hitResponse, ElectronicsSim* electronicsSim, bool addNoise);
```
Comment (civanch) [inline SimCalorimetry/EcalSimAlgos/interface/EcalTDigitizer.h:30]: @iarspider , template argument for this class is really used by consume class, I would not change this line.

File: SimCalorimetry/EcalSimAlgos/interface/EcalTDigitizer.h [hunk 2]
```diff
@@ -27,9 +27,9 @@ class EcalTDigitizer {
...
 
-  virtual ~EcalTDigitizer<Traits>();
+  virtual ~EcalTDigitizer();
```
Comment (civanch) [inline SimCalorimetry/EcalSimAlgos/interface/EcalTDigitizer.h:32]: this is OK to me.

File: SimCalorimetry/EcalSimAlgos/interface/EcalTDigitizer.h (hunk 1 above)
Comment (iarspider) [reply SimCalorimetry/EcalSimAlgos/interface/EcalTDigitizer.h:30]: @civanch the change is purely syntaxic ([CWG Issue 2337](https://cplusplus.github.io/CWG/issues/2237.html) ) - the `EcalTDigitizer` is still a template, the `Traits` typename is still available to all the functions inside this.

Comment (civanch) [reply SimCalorimetry/EcalSimAlgos/interface/EcalTDigitizer.h:30]: OK

**Comment by perrotta** [comment]
+1

**Pull request merged**